ml_perf/reference_implementation.py

```diff
@@ -91,10 +97,34 @@ def expand_flags(cmd, *args):
   # contain multiple instances of the same flag with different values.
   # Deduplicate, always taking the last occurance of the flag.
   deduped = OrderedDict()
+  deduped_vals = OrderedDict()
   for arg in expanded:
-    flag = arg.split('=', 1)[0]
-    deduped[flag] = arg
-  return deduped.values()
+    argsplit = arg.split('=', 1)
```

Comment:
multi_instance for reference_implementation.py itself?   The boolean that select whether multi-instance should be turn on could be done that way instead of in flags.  But the logic of decide how many instance to start better be automatic rather than specified, because the number of instance depends on total number of games and number of parallel games per instance.  The latter two better be set seperately for eval and selfplay.

Reply:
> multi_instance for reference_implementation.py itself? The boolean that select whether multi-instance should be turn on could be done that way instead of in flags. But the logic of decide how many instance to start better be automatic rather than specified, because the number of instance depends on total number of games and number of parallel games per instance. The latter two better be set seperately for eval and selfplay.

I agree. But you should put that logic directly into the `evaluate` and `selfplay` functions themselves rather than parsing the command line arguments in `expand_flags` or `checked_run`.